lib/src/create_report.dart

```diff
@@ -288,6 +294,122 @@ Future<ReportSection> _followsTemplate(
+          if (dependency is HostedDependency &&
+              !dependency.version.allows(latestVersion)) {
+            issues.add(_Issue(
+                'The constraint ${dependency.version} on $name does not support the latest published version $latestVersion',
+                span: _tryGetSpanFromYamlMap(pubspec.dependencies, 'name')));
+            dependenciesUpToDate = false;
+          }
+        }
+      }
+    }
+  } on Exception catch (e) {
+    issues.add(_Issue('Could not run pub outdated: $e'));
+    dependenciesUpToDate = false;
+  }
+
+  final currentSdkVersion = Version.parse(Platform.version.split(' ').first);
+  final sdkConstraint =
+      VersionConstraint.parse(pubspec.environment['sdk'] as String);
+  final allowsCurrentSdk = sdkConstraint.allows(currentSdkVersion);
+  if (!allowsCurrentSdk) {
+    _Issue(
+        'The current sdk constraint does not allow the latest Dart ($currentSdkVersion)',
+        span: _tryGetSpanFromYamlMap(pubspec.environment, 'sdk'),
+        suggestion: 'Try widening the upper boundary of the constraint.');
+  }
+
+  var allowsCurrentFlutter = true;
+  final flutterVersions = toolEnvironment.runtimeInfo.flutterVersions;
+  if (flutterVersions == null) {
+    issues.add(_Issue(
+        'Found no Flutter in your PATH. Could not determine the current Flutter version.'));
+    allowsCurrentFlutter = false;
+  } else {
+    final usesFlutter = await toolEnvironment.detectFlutterUse(packageDir);
+    final flutterDartVersion = Version.parse(
+        (flutterVersions['dartSdkVersion'] as String).split(' ').first);
+    final allowsCurrentFlutterDart = sdkConstraint.allows(flutterDartVersion);
+
+    if (!allowsCurrentFlutterDart) {
+      allowsCurrentFlutter = false;
+      issues.add(
+        _Issue(
+          'The current sdk constraint does not allow the dart version latest of Flutter ($flutterDartVersion)',
+          span: _tryGetSpanFromYamlMap(pubspec.environment, 'sdk'),
+        ),
+      );
+    } else {
+      if (usesFlutter) {
+        // TODO(sigurdm): this will not work well locally (installed version will
+        // not be latest). Perhaps we should query somewhere for the latest version.
+        final currentFlutterVersion =
+            Version.parse(flutterVersions['frameworkVersion'] as String);
+        final flutterConstraint =
+            VersionConstraint.parse(pubspec.environment['flutter'] as String);
+        if (!flutterConstraint.allows(currentFlutterVersion)) {
+          issues.add(
+            _Issue(
+              'The current flutter constraint does not allow the latest Flutter ($currentFlutterVersion)',
+              span: _tryGetSpanFromYamlMap(pubspec.environment, 'flutter'),
+            ),
+          );
+          allowsCurrentFlutter = false;
+        }
+      }
+    }
+  }
+  final publisher = json.decode(await read(
+      'https://pub.dev/api/packages/${pubspec.name}/publisher'))['publisherId'];
```

Comment:
Although it is a very unlikely attack vector on packages already published on pub.dev, let's escape `pubspec.name`.

Reply:
Done